tket/tests/Utils/test_Logging.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019-2022 Cambridge Quantum Computing
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Utils/TketLog.hpp>
+#include <catch2/catch.hpp>
+
+namespace tket {
+
+SCENARIO("Logging") {
+  tket_log()->set_level(LogLevel::Trace);
+
+  tket_log()->trace("Test trace (output).");
+  tket_log()->debug("Test debug (output).");
+  tket_log()->info("Test info (output).");
+  tket_log()->warn("Test warn (output).");
+  tket_log()->error("Test error (output).");
+  tket_log()->critical("Test critical (output).");
+
+  tket_log()->set_level(LogLevel::Off);
+
+  tket_log()->trace("Test trace (no output).");
+  tket_log()->debug("Test debug (no output).");
+  tket_log()->info("Test info (no output).");
+  tket_log()->warn("Test warn (no output).");
+  tket_log()->error("Test error (no output).");
+  tket_log()->critical("Test critical (no output).");
+
+  tket_log()->set_level(LogLevel::Err);
```

Comment:
is there anyway to capture the stdout stream and check correctness?

Reply:
Not that I could find, unfortunately: https://github.com/catchorg/Catch2/tree/devel/docs

Reply:
https://stackoverflow.com/questions/4810516/c-redirecting-stdout can we use this?

Reply:
It seems a lot easier to do this in pytest: I could add some tests there, along the lines of [this one](https://github.com/CQCL/tket/blob/a18e64f664a58c8d5b6d3bb8fe84b7a4ce9a9530/pytket/tests/measurement_setup_test.py#L68)?

Reply:
Alternatively: give the logging functions an optional argument representing an ostream, defaulting to `std::cout`; then the tests can use this. I might try that.

Reply:
yes I'm happy with either option, though the second sounds more thorough